platform-dom/src/main/java/com/softicar/platform/dom/elements/popup/configuration/DomPopupConfiguration.java

```diff
@@ -0,0 +1,290 @@
+package com.softicar.platform.dom.elements.popup.configuration;
+
+import com.softicar.platform.common.core.i18n.IDisplayString;
+import com.softicar.platform.common.core.interfaces.INullaryVoidFunction;
+import com.softicar.platform.common.core.interfaces.IStaticObject;
+import com.softicar.platform.dom.elements.popup.DomPopup;
+import com.softicar.platform.dom.elements.popup.DomPopupFrame;
+import com.softicar.platform.dom.elements.popup.position.strategy.DomPopupEventCoordinatesPositionStrategy;
+import com.softicar.platform.dom.elements.popup.position.strategy.DomPopupViewportCenterPositionStrategy;
+import com.softicar.platform.dom.elements.popup.position.strategy.DomPopupViewportOriginPositionStrategy;
+import com.softicar.platform.dom.elements.popup.position.strategy.IDomPopupPositionStrategy;
+import com.softicar.platform.dom.style.CssPercent;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * A mutable implementation of {@link IDomPopupConfiguration}.
+ *
+ * @author Alexander Schmidt
+ */
+public class DomPopupConfiguration implements IDomPopupConfiguration {
+
+	private IDisplayString caption;
+	private IDisplayString subCaption;
+	private DomPopupDisplayMode displayMode;
+	private IDomPopupPositionStrategy positionStrategy;
+	private INullaryVoidFunction callbackBeforeShow;
+	private INullaryVoidFunction callbackBeforeClose;
+	private boolean confirmBeforeClose;
+	private IStaticObject frameMarker;
+
+	/**
+	 * Creates a new instance, with sane defaults.
+	 */
+	public DomPopupConfiguration() {
+
+		this.caption = null;
```

Comment:
I was wondering that myself. Didn't wanna risk NPEs in case `null` captions are set for whatever reasons in the payload code. So left this unchanged. We should probably change that in a separate PR.

Reply:
pfff very ugly